Booker/wwwroot/js/site.js

```
@@ -4,65 +4,182 @@
 // Write your JavaScript code.
 
 function handleImageUpload(input) {
-    const preview = input.closest("section").querySelector("img");
-    const file = input.files[0];
-    const imageErrorSpan = input.closest("section").querySelector("small span");
+    const preview = input.closest("section").querySelector(".image-preview-container");
 
-    if (file) {
-        const reader = new FileReader();
-        reader.onload = function (e) {
-            const img = new Image();
-            img.onload = function () {
-                const MAX_WIDTH = 800;
-                const MAX_HEIGHT = 600;
-                let width = img.width;
-                let height = img.height;
-
-                if (width > height) {
-                    if (width > MAX_WIDTH) {
-                        height *= MAX_WIDTH / width;
-                        width = MAX_WIDTH;
-                    }
-                } else {
-                    if (height > MAX_HEIGHT) {
+    const imageErrorSpan = input.closest("section").querySelector("small span");
+    const files = Array.from(input.files).slice(0, 6);
```

Personally, I would explicitly reject uploading more than 6 files (or whatever the number) with an error message rather than silently truncate the array.